### ISSUE TITLE
windowManager.i3.config = null no longer works

### ISSUE THREAD

**Issue by wentasah** [nix-community/home-manager#1985]
### Issue description

When `home.nix` contains

    windowManager.i3.config = null;

then `home-manager switch` fails with

    error: value is null while a set was expected, at .../modules/services/window-managers/i3-sway/i3.nix:262:36

This seems to have been broken by #1980. I'm trying to fix it. Will send PR soon.

#### Maintainer CC

@sumnerevans @alarsyo

**Comment by sumnerevans** [comment]
In your PR, can you please also add a test to ensure that we don't break that again?

**Comment by wentasah** [comment]
I've never written a test for home manager. It looks like I have to add a file to `tests/modules/services/window-managers/i3` and reference it from `default.nix`. Is that correct?

**Comment by sumnerevans** [comment]
Yes, that's correct. Then you use the `nmt.script` to define what the test should do.
